core/string/encode_spec.rb

```diff
@@ -60,6 +60,12 @@
 
       "\rfoo".encode(universal_newline: true).should == "\nfoo"
     end
+
+    it "replaces invalid encoding" do
+      invalid = "\xe3\x81\xa1\xE3\x81\xFF".force_encoding("UTF-8")
```

Comment:
No need to `force_encoding`, it's the encoding for all String literals per the magic encoding comment at the top.